src/main/java/io/statnett/k3a/lagexporter/ClusterLagCollector.java

```diff
@@ -231,4 +237,18 @@ private void invalidateClients() {
         }
     }
 
+    private static boolean isTimeout(final Throwable t) {
+        Throwable current = t;
+        while (current != null) {
+            if (TimeoutException.class.isAssignableFrom(current.getClass())) {
+                return true;
+            }
+            if (current == current.getCause()) {
```

Comment:
Is this required? Looks a bit suspicious.

Reply:
The `TimeoutException` we are looking for sometimes gets wrapped in an `ExecutionException`.

Reply:
Strange...